### PR TITLE
Adding canonical examples for Dough components

### PR DESCRIPTION
**Pull request by benbarnett**
Where possible, this will link to the best examples of where Dough components are being used both for implementation and rendered in the browser.

### CODE REVIEW

**Comment by rnalexander** [comment]
:+1:

**Comment by neoeno** [comment]
Neat! Feel like it'd be cool to see code examples directly — something like [Bootstrap's](http://getbootstrap.com/components/#jumbotron). I guess you did it this way for a reason though?

**Comment by benbarnett** [comment]
@neoeno have you seen http://madewithdough.org.uk/? It pulls these docs through and renders the code samples.

If not, perhaps we need another list to list the things that list the things.

**Comment by neoeno** [comment]
Ah yes — I do remember that actually, sorry! In that case this is a great addition — to also have reference to examples in the wild. It'd be good to have examples of the more complicated stuff too — the JS stuff is well documented as JS, but markup examples take a little bit of finding. `Validation` & `FieldHelpText`, etc.

There's some other great stuff spread between dough and yeast that sometimes doesn't get reused as much as it could, because you have to know it's there before you look for it. Particularly a lot of the layouts stuff, forms stuff — it's easy to forget the exact structure. The [styleguide page](https://www.moneyadviceservice.org.uk/en/styleguide/forms) is good for that, but doesn't reference the helpers.

Anyway, that's a different issue. Not sure if you're looking for thumbs because this is a WIP! But :+1:

File: docs/ButtonPrimary.md [hunk 1]
```diff
@@ -3,3 +3,8 @@
 * Primary / next action button
 * Can have more than one, providing they are all next actions
 * For example if there is a 'Save' page that asks the user to Login or Register, they should both be yellow as they are the same prominence
+
+### Examples
+
+* [Code example](https://github.com/moneyadviceservice/rad/blob/3028f9cf236f2d7c45bea02c3d1d2b47534d3424/app/views/principals/new.html.erb#L100)
+* [Live example](https://radsignup.moneyadviceservice.org.uk/)
```
Comment (munckymagik) [inline docs/ButtonPrimary.md]: Hmmmm, RAD is still growing up, I wonder if we should link to a more mature project where things are less likely to change?

Comment (benbarnett) [inline docs/ButtonPrimary.md]: Fair point. The only other example I can see is on the cutback calculator, maybe we'll use that? https://github.com/search?q=user%3Amoneyadviceservice+button_primary&type=Code&utf8=%E2%9C%93

**Comment by munckymagik** [comment]
Just wanted to say other than my comment about RAD :+1:

**Pull request merged**